sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -40,10 +42,16 @@ private[sql] class SharedState(val sparkContext: SparkContext) {
    */
   val listener: SQLListener = createListenerAndUI(sparkContext)
 
+  lazy val hadoopConf: Configuration = {
+    val conf = sparkContext.hadoopConfiguration
+    conf.addResource(Utils.getContextOrSparkClassLoader.getResource("hive-site.xml"))
```

Comment:
There is no fundamental difference between these two. After we call `conf.addResouorce("hive-site.xml")`, `Configuration` internally will use classloader's `getResource` to find the url of the resource. At here, `Utils.getContextOrSparkClassLoader.getResource("hive-site.xml")` just explicitly get the url.